pkg/controller/keycloakbackup/keycloakbackup_controller.go

```diff
@@ -120,6 +120,11 @@ func (r *ReconcileKeycloakBackup) Reconcile(request reconcile.Request) (reconcil
 	if err != nil {
 		return r.ManageError(instance, err)
 	}
+	if len(keycloaks.Items) == 0 {
+		err = fmt.Errorf("backup %v/%v has a matching Keycloak cluster is down and will be ignored", instance.Namespace, instance.Name)
+		log.Info(err.Error())
+		return r.ManageError(instance, err)
+	}
```

Comment:
There is an alternative solution here. You could return an error from `GetMatchingKeycloaks` if the list is empty there. This would mean a single change to that function vs the 3 identical changes here.

Reply:
I totally agree! That's a very good idea!